### ISSUE TITLE
 sparsity_loss = entropy

### ISSUE THREAD

**Issue by wtj-zhong** [yashbhalgat/HashNeRF-pytorch#38]
When I run this code with the synthetic Lego dataset, it works fine. But when I run it with the llff dataset, I encounter the following issue:
python run_nerf.py --config configs/fren.txt --finest_res 512 --log2_hashmap_size 19 --lrate 0.01 --lrate_decay 10 0 0.0010004043579101562 [00:00<?, ?it/s] [1/1]
[99%]
c:\users\nezo\desktop\3d\hashnerf-pytorch\run_nerf.py(379)raw2outputs() -> 
sparsity_loss = entropy
Could you please tell me the reason for this issue?

**Comment by Fjzd** [comment]
I also encountered this error, and I believe it was (1.0 weights. sum (-1, keepdim=True))<0 that caused the entropy() to fail.You could normalize the weights to correct the error.

**Comment by THUROI0787** [comment]
I had the same problem. I'm not sure why, but I retrained without changing anything, and it worked. May be a problem in the initialization and optimization of neural network parameters, with randomness?

**Comment by zshuai9508** [comment]
I also encountered the same problem on the self-made data set. Is there something wrong with the data?

**Comment by Fjzd** [comment]
I think the problem is caused by  weights = (x - voxel_min_vertex)/(voxel_max_vertex-voxel_min_vertex) of Hash_encoding!You can change it into  weights = (x - voxel_min_vertex)/(voxel_max_vertex-voxel_min_vertex+1e-6) and try it again!

**Comment by wuzuyin** [comment]
May I ask everyone, why do we add this loss function and what is its purpose?
![image](https://github.com/yashbhalgat/HashNeRF-pytorch/assets/52769308/bbaf54f3-f941-4d4e-8b3f-81c529fe877e)
